manifests/named_instance.pp

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) 2018 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+
+# A named instance
+#
+# @example
+define nebula::named_instance(
+  String            $path,
+  Integer           $uid,
+  Integer           $gid,
+  String            $fauxpaas_pubkey,
+  String            $fauxpaas_puma_config,
+  Array[String]     $users = [],
+  Array[String]     $subservices = [],
+) {
+
+  # Create the application user's group
+  group { $title:
+    ensure => 'present',
+    gid    => $gid,
+  }
+
+  # Add sudoers and passed users to the group
+  (lookup('nebula::users::sudoers').map |$user, $_data| { $user } + $users).each |$user| {
+    exec { "${user} ${title} membership":
+      unless  => "/bin/grep -q ${title}\\S*${user} /etc/group",
+      onlyif  => "/usr/bin/id ${user}",
+      command => "/sbin/usermod -aG ${title} ${user}",
```

Comment:
Our users really should be virtual resources, but I think that's out of scope for this ticket.